### ISSUE TITLE
Insufficient memory when evaluating model

### ISSUE THREAD

**Issue by Passenger12138** [facebookresearch/detr#423]
## Instructions To Reproduce the Issue:

1.I try adapt the model to my Ubuntu server
2. I follow the instructions of the readmeI to evaluate DETR R50 on COCO val5k but with a single CPU run:，Here are the specific commands I ran。
```
python3 detr/main.py --batch_size 2 --device cpu --num_workers 0 --no_aux_loss --eval --resume https://dl.fbaipublicfiles.com/detr/detr-r50-e632da11.pth --coco_path coco_2017/
```
3. What puzzles me is that the evaluation model occupies too much memory. I allocated 64G of memory. When running the evaluation model to [90/2500], it already occupies more than 65% of the memory. At the same time, its memory footprint is still With the constant increase, what I want to know is whether this evaluation mode must occupy so much memory or there are some mistakes and how can I modify it？（The following is the specific display picture）
![image](https://user-images.githubusercontent.com/47623700/129162172-68fbcc98-5909-4796-9510-f613263f89e7.png)
![d546c71caa6208639da15f6520d2fad](https://user-images.githubusercontent.com/47623700/129161871-e674fe93-40e5-4c86-a34d-195859b861da.png)

## Environment:

This is my server environment configuration
```
python -m torch.utils.collect_env
```
![image](https://user-images.githubusercontent.com/47623700/129160269-20c3cb0e-83ed-4fc8-889c-62c7252d32e8.png)

**Comment by Passenger12138** [comment]
Because there is no _new_empty_tensor() function in torchvision0.2.1, I rewrite the implementation of the function in torchvision0.6.0 in torchvision0.2.1. The function is based on the torch.ops.torchvision._new_empty_tensor_op() function, which is implemented in torchvision0. Not in version 2.1. The implementation in version 0.6.0 of torchvision involves _register_onnx_ops.py and vision.cpp files, so it is difficult for me to implement this function.

**Comment by fmassa** [comment]
@Passenger12138 latest PyTorch has support for tensors with 0 elements in its batch dimension.

So you can basically replace this whole block https://github.com/facebookresearch/detr/blob/eb9f7e03ed8e2ed2cd55528989fe7df890bc3fc0/util/misc.py#L450-L467

with
```python
interpolate = torch.nn.functional.interpolate
```

**Comment by fmassa** [comment]
Note that the code as it currently stands should already work with latest torchvision, since https://github.com/facebookresearch/detr/pull/404 was merged

**Comment by Passenger12138** [comment]
> 请注意，由于#404已合并，目前的代码应该已经适用于最新的 Torchvision

Thank you very much for your suggestions, but for some reason, I have to use (pytorch 1.3.0 and torchvision 0.2.1), so could I simply rewrite this function?

**Comment by fmassa** [comment]
I'm not sure DETR will work on PyTorch 1.3 and torchvision 0.2.1

**Comment by Passenger12138** [comment]
Thanks for your reply, maybe you can close this question now
